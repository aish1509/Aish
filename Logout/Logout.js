import React from 'react';
import PropTypes from 'prop-types';
import './Logout.scss';

import {makeStyles} from '@material-ui/styles';

export const useStyles = makeStyles(theme => ({
	container: {
		height: '50px',
		display: 'flex',
		justifyContent: 'center',
		alignContent: 'center',
		alignItems: 'flex-end',
		flexDirection: 'column',
		fontFamily: [
			'OpenSans-Regular',
			'Roboto',
			'Helvetica',
			'Arial',
			'sans-serif'
		],
		fontSize: '14px',
		color: theme.colors.blueGrey[20],
		marginRight: '10px'
	},
	welcome: {
		fontFamily: [
			'OpenSans-Regular',
			'Roboto',
			'Helvetica',
			'Arial',
			'sans-serif'
		],
		fontSize: '10px',
		color: theme.colors.blueGrey[100]
	},
	logout: {
		fontFamily: [
			'OpenSans-Regular',
			'Roboto',
			'Helvetica',
			'Arial',
			'sans-serif'
		],
		fontSize: '12px',
		color: theme.colors.blueGrey[20],
		border: 'none',
		textDecorationLine: 'underline',
		background: 'none',
		padding: '0'
	}
}));

const Logout = ({firstName, onLogout}) => {
	const classes = useStyles();

	const handleClick = e => {
		e.preventDefault();
		onLogout();
	};

	return (
		<div className={classes.container}>
			Welcome, {firstName}!
			<button
				className={classes.logout}
				type="button"
				onClick={handleClick}
			>
				Logout
			</button>
		</div>
	);
};

Logout.propTypes = {
	firstName: PropTypes.string.isRequired,
	lastName: PropTypes.string.isRequired,
	onLogout: PropTypes.func.isRequired
};

export default Logout;
