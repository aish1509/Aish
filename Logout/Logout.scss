.logout {
  width: 140px;
  height: 100px;
  padding: 5px 5px;
  background-color: gray;
  color: white;
  font-family: 'Helvetica Neue', Helvetica, Arial, sans-serif;
  font-weight: bold;
  font-size: 13px;
}

.logout__display {
  padding-top: 8px;
}

.logout__display__seperator {
  padding: 5px;
  font-weight: normal;
}

.logout__display__link {
  background-color: transparent;
  color: white;
  text-decoration: underline;
  font-weight: bold;
  font-size: 13px;
  font-family: Arial, 'Helvetica Neue', Helvetica, sans-serif;
  border: none;
  cursor: pointer;
  outline: none;
}
