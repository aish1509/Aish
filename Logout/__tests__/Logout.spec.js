import React from 'react';
import {shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';
import Logout, {useStyles} from '../Logout';

describe('<Logout />', () => {
	let wrapper = null;
	let mockOnLogout = null;
	let mockEvent = null;

	beforeEach(() => {
		mockOnLogout = jest.fn();
		mockEvent = {
			preventDefault: jest.fn()
		};
		wrapper = shallow(
			<Logout lastName="Doe" firstName="John" onLogout={mockOnLogout} />
		);
	});

	it('should render correctly', () => {
		expect(shallowToJson(wrapper)).toMatchSnapshot();
	});

	it('handleClick', () => {
		wrapper.find('button').simulate('click', mockEvent);
		expect(mockEvent.preventDefault).toHaveBeenCalled();
		expect(mockOnLogout).toHaveBeenCalled();
	});

	it('styles matches snapshot', () => {
		expect(useStyles()).toMatchSnapshot();
	});
});
