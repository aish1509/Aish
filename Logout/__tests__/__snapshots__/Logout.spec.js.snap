// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logout /> should render correctly 1`] = `
<div
  className={
    Object {
      "alignContent": "center",
      "alignItems": "flex-end",
      "color": "#E6F0F7",
      "display": "flex",
      "flexDirection": "column",
      "fontFamily": Array [
        "OpenSans-Regular",
        "Roboto",
        "Helvetica",
        "Arial",
        "sans-serif",
      ],
      "fontSize": "14px",
      "height": "50px",
      "justifyContent": "center",
      "marginRight": "10px",
    }
  }
>
  Welcome, 
  John
  !
  <button
    className={
      Object {
        "background": "none",
        "border": "none",
        "color": "#E6F0F7",
        "fontFamily": Array [
          "OpenSans-Regular",
          "Roboto",
          "Helvetica",
          "Arial",
          "sans-serif",
        ],
        "fontSize": "12px",
        "padding": "0",
        "textDecorationLine": "underline",
      }
    }
    onClick={[Function]}
    type="button"
  >
    Logout
  </button>
</div>
`;

exports[`<Logout /> styles matches snapshot 1`] = `
Object {
  "container": Object {
    "alignContent": "center",
    "alignItems": "flex-end",
    "color": "#E6F0F7",
    "display": "flex",
    "flexDirection": "column",
    "fontFamily": Array [
      "OpenSans-Regular",
      "Roboto",
      "Helvetica",
      "Arial",
      "sans-serif",
    ],
    "fontSize": "14px",
    "height": "50px",
    "justifyContent": "center",
    "marginRight": "10px",
  },
  "logout": Object {
    "background": "none",
    "border": "none",
    "color": "#E6F0F7",
    "fontFamily": Array [
      "OpenSans-Regular",
      "Roboto",
      "Helvetica",
      "Arial",
      "sans-serif",
    ],
    "fontSize": "12px",
    "padding": "0",
    "textDecorationLine": "underline",
  },
  "welcome": Object {
    "color": "#4C5D65",
    "fontFamily": Array [
      "OpenSans-Regular",
      "Roboto",
      "Helvetica",
      "Arial",
      "sans-serif",
    ],
    "fontSize": "10px",
  },
}
`;
